### ISSUE TITLE
could not create directory when updating theme

### ISSUE THREAD

**Issue by 2nickpick** [afragen/github-updater#307]
I've successfully set up a private Bitbucket theme in the Giithub Updater settings, and it installs and activates the theme perfectly. 

Then I force an update by reducing the version number in my local `style.css`. 

When I attempt the update I get the following: 

```
Downloading update from https://bitbucket.org/**user**/**theme**/get/master.zip…

Unpacking the update…

Installing the latest version…

Enabling Maintenance mode…

Renaming **user**-**theme-name**-a88bb7670f98 to **theme-name**…

Rename successful…

Removing the old version of the theme…

Could not create directory. /www/wp-content/themes/**theme-name**/

Theme update failed
```

The permissions on my self-hosted server are set to let the entire WP installation be writable. Not sure the next steps to take.

**Comment by afragen** [comment]
Try adding `define('FS_METHOD', 'direct');` to your wp-config.php file.

**Comment by 2nickpick** [comment]
That worked.... but is it generally safe for production?

**Comment by afragen** [comment]
Not sure I can answer that question.

**Comment by 2nickpick** [comment]
ok, thanks. You've got me in the right direction.

**Comment by afragen** [comment]
Please report back what worked for you. Thanks.

**Comment by 2nickpick** [comment]
It's pretty strange - I set `FS_METHOD` to `direct` initially and the theme would update. 

Replaced it with the `FS_CHMOD_FILE` and `FS_CHMOD_DIR` from above with fewer permissions.  Still worked great. 

Removed everything altogether, still worked! I think I may have just not had my file permissions set up correctly from the get go, and after toying around with the FS defines I may have tweaked the permissions on the server further to get everything in place. 

I'm not able to replicate the problem I was having earlier. 

In case anyone wants more info on these defines, they are mentioned in the WP Codex: 

https://codex.wordpress.org/Editing_wp-config.php#Override_of_default_file_permissions
